src/commonMain/kotlin/uuid.kt

```diff
@@ -94,12 +94,51 @@ public expect val Uuid.variant: Int
 public expect val Uuid.version: Int
 
 /**
- * Construct new [Uuid] instance from a [String]
+ * Parse this string as [UUID][Uuid].
  *
- * @param from The [String] representation of the UUID
+ * Only the formal string representation as defined in RFC&nbsp;4122 is
+ * supported (e.g. `f81d4fae-7dec-11d0-a765-00a0c91e6bf6`). The hexadecimal
+ * digits `a` through `f` can be either lowercase or uppercase.
+ *
+ * @see <a href="https://tools.ietf.org/html/rfc4122#section-3">RFC 4122: Section 3</a>
+ * @throws IllegalArgumentException if parsing fails.
+ */
+// Implementation Notes:
+// - We define this as an extension function on string to simplify null handling
+//   for our users: `value?.let(::uuidOf)` vs `value?.toUuid()`
+//   TODO should we provide both an inline one of them?
+//
+// @ExperimentalStdlibApi
+// @SinceKotlin("1.x")
+public expect fun String.toUuid(): Uuid
```

Comment:
Why not single argument `uuidOf`?

Reply:
For the reasons given in the comment: `value?.let(::uuidOf)` is not as nice as `value?.toUuid()`. We could provide both and _inline_ one of them away.